### PR TITLE
Bug 1408100 - Upgrade to Bootstrap 4

### PR DESCRIPTION
**Pull request by camd**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,10 +40,23 @@
     "neutrino-preset-react": "4.2.3",
     "ngreact": "0.4.1",
     "numeral": "2.0.6",
+    "popper.js": "^1.12.5",
     "raw-loader": "0.5.1",
-    "react": "15.3.2",
-    "react-dom": "15.3.2",
-    "taskcluster-client": "2.5.0"
+    "react": "^15.4.2",
```
Comment (edmorley) [inline package.json]: Don't forget to remove these react changes and other package updates from this PR :-)

**Comment by wlach** [comment]
Have you tested this how Perfherder works with the updated dependency?

**Comment by camd** [comment]
@wlach : Yeah.  I'm still in progress on this.  But the short answer is: Yes, I've been testing it as I made changes.  But not done with all the changes yet...

One of the big things is that I need to remove dependency on ``angular-ui-bootstrap`` which is heavily used for modal dialogs.  But ``angular-ui-bootstrap`` only supports Bootstrap 3.x and definitely DOESN'T work with 4.  I am changing those to be angular ``components`` and Bootstrap-4 modals without ``angular-ui-bootstrap``.  That will come in another PR to precede this one.

So, yeah, this will heavily affect Perfherder.  I'm testing it as best I can as I go.  But I was also hoping to hit you up later this week to perhaps help a bit with running through some scenarios.  

This has been a long process since it largely affects almost EVERY part of our UI.  Quite a slog, but I think it will open doors for us long-term.

**Comment by wlach** [comment]
Ok good to know. Some parts of perfherder use angular-ui-bootstrap rather extensively (I think moreso than treeherder, actually) so was a bit worried about that angle in particular. Feel free to hit me up anytime if you want to discuss further, I'll be around.

**Comment by camd** [comment]
Closing this one as I took a different tack in another branch in PR #2883

**Pull request closed without merging**